### ISSUE TITLE
Address tag language display for editing

### ISSUE THREAD

**Issue by mmaker-gh** [otoDB/otoDB#260]
Need to do something about this since it will hinder accessibility for users who are unable to read a (base) tag's language:
<img width="833" height="96" alt="Image" src="https://github.com/user-attachments/assets/3af449f4-bd03-43b7-848b-d89eff35babd" />

The options I see for this are:
- A: Enforce using English for the base tag whenever possible
	- Simplest solution
	- Not preferred, problematic for non-English speakers
- B: Use user's language preference and display the corresponding tag/slug
	- Bit more effort but better solution imo
	- Might require some clean-up to how slugs are processed so they're more readable (things like parenthesis get removed and some have `_1` suffixed which gets confusing)
		- Can we avoid using slugs here...?
	- Maybe do this in conjunction with option A?
